src/Interpreters/executeQuery.cpp

```diff
@@ -2025,6 +2042,10 @@ void executeQuery(
             pipeline.setProgressCallback(context->getProgressCallback());
         }
 
+        /// input stream has been consumed into some source proceccors/format readers
+        /// here we reset it in order not to hold the reference to the input stream
+        istr.reset();
```

Comment:
I suppose it should be nullptr after 

```
streams = executeQueryImpl(begin, end, context, flags, QueryProcessingStage::Complete, istr, ast, implicit_tcl_executor);
```

will assert be more suitable here?

Reply:
Lets try